### DISCUSSION TITLE
app_commands before_invoke & after_invoke

### DISCUSSION THREAD

**Discussion by murtadapy** [Rapptz/discord.py#9212] [General]
Hi,

Is there a way to use a similar functionality of before_invoke and after_invoke on app_commands? Are we able to create custom decorators on commands that have interaction param?

Thanks!

**Comment by Soheab** [comment]
This discussion answer might be able to help with this https://github.com/Rapptz/discord.py/discussions/8372#discussioncomment-3459014

**Reply by murtadapy** [reply]
That worked, thank you so much!
